app/models/miq_request_workflow.rb

```diff
@@ -1042,7 +1042,8 @@ def ems_folder_to_hash_struct(ci)
   end
 
   def storage_to_hash_struct(ci)
-    build_ci_hash_struct(ci, [:name, :free_space, :total_space, :storage_domain_type])
+    storage_cluster = ci.storage_clusters.detect { |cluster| cluster.ems_id == get_source_and_targets[:ems].try(:id) }
+    build_ci_hash_struct(ci, [:name, :free_space, :total_space, :storage_domain_type]).tap { |hs| hs.storage_cluster = storage_cluster.try(:name) }
```

Comment:
some minor style nitpicks

```ruby
ems_id = get_source_and_targets[:ems].try(:id)
storage_cluster = ci.storage_clusters.detect { |cluster| cluster.ems_id == ems_id }
cluster_name = storage_cluster.try(:name)
build_ci_hash_struct(ci, [:name, :free_space, :total_space, :storage_domain_type]).tap { |hs| hs.storage_cluster = cluster_name }
```

I like building up the code with descriptive variable name. I know its less dense, but imho makes it more readable for a future developer and might serve up for documentation.
Also you can check the edge cases, in case name is nil or blank?
And this uses less method calls, as both name and id always return the same